pkg/controllers/management/kontainerdrivermetadata/metadata_handler.go

```diff
@@ -69,55 +74,91 @@ func Register(ctx context.Context, management *config.ManagementContext) {
 		ctx:                       ctx,
 	}
 
-	// load values on startup and start ticker
-	url, interval, err := GetSettingValues()
-	if err == nil {
-		if err := m.Refresh(url, true); err != nil {
-			logrus.Errorf("driverMetadata failed to refresh %v", err)
-		}
-		m.initTicker(ctx, interval)
-	}
-
 	mgmt.Settings("").AddHandler(ctx, "rke-metadata-handler", m.sync)
 }
 
-func (m *MetadataController) sync(key string, test *v3.Setting) (runtime.Object, error) {
-	if test == nil || (test.Name != "rke-metadata-url" && test.Name != "rke-metadata-refresh-interval-minutes") {
+func (m *MetadataController) sync(key string, setting *v3.Setting) (runtime.Object, error) {
+	if setting == nil || (setting.Name != rkeMetadataURL && setting.Name != rkeMetadataInterval) {
 		return nil, nil
 	}
-	url, interval, err := GetSettingValues()
-	if err != nil {
-		return nil, fmt.Errorf("driverMetadata error getting settings %v", err)
-	}
-	if tickerData != nil {
-		if err := m.Refresh(url, false); err != nil {
-			return nil, fmt.Errorf("driverMetadata failed to refresh %v", err)
+
+	var interval time.Duration
+	if tickerData == nil {
+		if err := m.initTicker(); err != nil {
+			return nil, err
 		}
-		if tickerData.interval != interval {
-			tickerData.cancelFunc()
+		return setting, nil
+	}
 
-			logrus.Infof("driverMetadata: starting new %v", interval.Minutes())
-			cctx, cancel := context.WithCancel(m.ctx)
-			tickerData.interval = interval
-			tickerData.cancelFunc = cancel
+	if setting.Name == rkeMetadataURL {
+		if err := m.refresh(); err != nil {
```

Comment:
We should do m.refresh() only if the URL setting is updated than the previous URL